docs/docs/reference/gen_notebooks/weave_via_service_api.md

```diff
@@ -0,0 +1,488 @@
+---
+title: Use the Service API to Log and Query Traces
+---
+
+
+:::note[This is a notebook]
+
+<a href="https://colab.research.google.com/github/wandb/weave/blob/master/docs/./notebooks/weave_via_service_api.ipynb" target="_blank" rel="noopener noreferrer" class="navbar__item navbar__link button button--secondary button--med margin-right--sm notebook-cta-button"><div><img src="https://upload.wikimedia.org/wikipedia/commons/archive/d/d0/20221103151430%21Google_Colaboratory_SVG_Logo.svg" alt="Open In Colab" height="20px" /><div>Open in Colab</div></div></a>
+
+<a href="https://github.com/wandb/weave/blob/master/docs/./notebooks/weave_via_service_api.ipynb" target="_blank" rel="noopener noreferrer" class="navbar__item navbar__link button button--secondary button--med margin-right--sm notebook-cta-button"><div><img src="https://upload.wikimedia.org/wikipedia/commons/9/91/Octicons-mark-github.svg" alt="View in Github" height="15px" /><div>View in Github</div></div></a>
+
+:::
+
+<!--- @wandbcode{service-api-colab} -->
+
+In the following notebook, you will learn how to use the Weave Service API to log traces. Specifically, you will use the Service API to:
```

Comment:
We have a first-party `trace_server_interface` for this -- curious why you're using the raw HTTP endpoint instead of that?

Reply:
The initial code was created by @m-rgba - maybe they can provide that context? I just documented the code as it was. @andrewtruong do you think it would be promoting an anti pattern by using the raw endpoint? If so, I can go back in and update the notebook to use trace_server_interface?

Reply:
Hey @andrewtruong - so when I initially created it:
- I made them while I was making a OpenWebUI plugin where I didn't want the user to have to install any packages.
- A customer then had asked for some examples using the REST API since they were having trouble with the client.

My argument for why I feel like this could be valuable:
- It's a good escape-hatch for anyone running where they can't use our client, or are having trouble with our client, but they want to send us traces.
- The service API is the most language neutral method for sending us traces.

If either of these are true, our current OpenAPI docs get the user like 95% there - but are pretty easy to flesh out some parts like - what's the shape of a completion we're expecting, or how should they format token counts.

100% open to this being an anti-pattern though, if we're going to be updating the API a lot or if there are other reasons from the product-adoption side. 
It also may be a both kind of thing where we can have examples for manually creating traces with the client as well.

Reply:
For python we should recommend the first-party stuff, and for other languages we can recommend this HTTP approach.

That said, I'm not against it.  We can still show python using the service API